### PR TITLE
Replacing master with main (SCP-3446)

### PR DESCRIPTION
**Pull request by bistline**
This renames the master branch to main and replaces any references with the new name.  This will affect CD only for production deployments, but as the branch will be renamed before any deployments happen this should not matter.

You will need to run the following commands to update your local repository:

```
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a 
```

### CODE REVIEW

**Review by jlchang — APPROVED** [review]
Changes make sense to me! Thanks for transitioning us from `master` to `main`.

File: .github/workflows/delete-branch-on-merge-to-master.yml
Comment (eweitz) [inline .github/workflows/delete-branch-on-merge-to-master.yml:1]: Renaming this file seems clearest.  Preserving Git history across renames is tricky; `git mv` is my go-to.  If history isn't preserved, I think that's OK here.  Preserving clarity via the rename seems more important.

`git mv .github/workflows/delete-branch-on-merge-to-master.yml .github/workflows/delete-branch-on-merge-to-main.yml`

File: .github/workflows/delete-branch-on-merge-to-master.yml [hunk 1]
```diff
@@ -1,16 +1,16 @@
 name: Selectively delete feature branch after merge to master
```
Comment (eweitz) [inline .github/workflows/delete-branch-on-merge-to-master.yml:1]:
```suggestion
name: Selectively delete feature branch after merge to main
```

**Review by eweitz — APPROVED** [review]
Nice update to modern defaults!  

I reviewed https://github.com/search?q=repo%3Abroadinstitute%2Fsingle_cell_portal_core%20master&type=code and found a few missed references that would help to update, one noted here and two in comments below.

* [Line 34 in search_facet.rb](https://github.com/broadinstitute/single_cell_portal_core/blob/db53124dd325b7047cefbb1c04b9c54f306d54cf/app/models/search_facet.rb#L34)

I also replaced `master` with `main` in the [SCP release engineer](https://docs.google.com/document/d/1rKcOQPtkbhZCbTqxTnaT-4E29VLzkJrm0kAoWjLx3QA/edit) docs.  The [SCP playbook](https://docs.google.com/document/d/1vJz6hXdjAu8LBAfIS4oMy7yLIUX177gU1H7BxTPPzAU/edit) has more references that would be good to review and change as warranted.

Do any of our Terraform integrations need to be updated?

**Pull request merged**